assets/js/types/type-defs/payments.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+export type ObjectType = Record< string, unknown >;
+export type EmptyObjectType = Record< string, never >;
+
+/**
+ * Internal dependencies
+ */
+import type { CartTotals } from './cart';
+import {
+	CartResponseBillingAddress,
+	CartResponseShippingAddress,
+} from './cart-response';
+
+export interface Supports {
+	showSavedCards?: boolean;
+	showSaveOption?: boolean;
+	features?: string[];
+	// Deprecated, in favour of showSavedCards and showSaveOption
+	savePaymentInfo?: boolean;
+}
+
+// we assign a value in the class for supports.features
+export interface SupportsInConstructor extends Supports {
+	features: string[];
+}
+
+export interface CanMakePaymentArgument {
+	cartTotals: CartTotals;
+	cartNeedsShipping: boolean;
+	billingData: CartResponseBillingAddress;
+	shippingAddress: CartResponseShippingAddress;
+	selectedShippingMethods: Array< unknown >;
+	paymentRequirements: Array< string >;
+}
+
+export type CanMakePayment = ( cartData: CanMakePaymentArgument ) => boolean;
+
+export interface PaymentMethodConfig {
```

Comment:
Possibly rename this to `PaymentMethodConfigurationSettings` or something to make it clearer that this is not the same thing as the class?